packages/text-field/Input.js

```diff
@@ -43,20 +40,44 @@ export default class Input extends React.Component {
   }
 
   componentDidUpdate(prevProps) {
+    const {
+      id,
+      handleValueChange,
+      setInputId,
+      setDisabled,
+      foundation,
+      value,
+      disabled,
+    } = this.props;
+
     this.handleValidationAttributeUpdate(prevProps);
 
-    if (this.props.disabled !== prevProps.disabled) {
-      const {disabled} = this.props;
-      this.props.setDisabled(disabled);
-      this.props.foundation.setDisabled(disabled);
+    if (disabled !== prevProps.disabled) {
+      setDisabled(disabled);
+      foundation.setDisabled(disabled);
     }
 
-    if (this.props.id !== prevProps.id) {
-      this.props.setInputId(this.props.id);
+    if (id !== prevProps.id) {
+      setInputId(id);
+    }
+
+    // this should be in the componentDidMount, but foundation is not created
+    // at that time.
+    if (value && foundation !== prevProps.foundation) {
```

Comment:
Once this https://github.com/material-components/material-components-web-react/pull/353 is fixed, I will move this back into componentDidMount